enhancements/kube-apiserver/enforce-labels.md

```diff
@@ -0,0 +1,130 @@
+])
+- seem to be interested in adding a updates to existing resources in reaction to MutationTemplate changes (but no hard commitment on this)
+- a regularly release opensource project with  community backed by companies like google, ms, RH, ...
+- multiple Red Hat teams want to have this as part of Openshift
+
+CONs:
+- don't provide any functionality we need yet
+- reactive resource updates still under design
+- alpha version of the mutation webhook only available by the end of the year
+- an overkill for our immediate needs, provides a much broader functionality than we need
+
+NOTE: We are only interested in the planned mutating feature of the project.
+
+#### Gatekeeper operator
+
+Gatekeeper would be installed using an operator. The operator would not only take care of the installation, but also perform some additional configuration to ensure gatekeeper does not interere with control-plane resources. 
+
+There is currently an effort to provide a gatekeeper operator [here|https://github.com/font/gatekeeper-operator]. It is planned that the operator will be accepted as part of the open-policy-agent repo.
+
+Gatekeeper inspects resources based on a validating webhook (we assume the mutation feature will similarly use a mutating webhook). A sample webhook definition can be found [here|https://github.com/open-policy-agent/gatekeeper/blob/1de87b6d3c2ed3609e69b789d722e28285873861/charts/gatekeeper/templates/gatekeeper-validating-webhook-configuration-validatingwebhookconfiguration.yaml]. The operator would modify the webhook namespace selector to exclude all control-plane components. The operator would keep track of the namespaces and only allow the gatekeeper webhooks access to non control-plane namespaces. The operator configuration could allow to configure the namespaces accessible by gatekeeper.
+
+#### Safeguaring control-plane components
+
+Gatekeeper uses a single admission webhook to for constraints enforcement. We assume the same solution will be used for the mutation feature. The control-plane resources can be excluded from gatekeeper interference by specifying an appropriate namespace selector on the webhook.
+Gatekeeper would be installed/configured by an operator, which would adjust the webhook namespace selector (based on some GatekeeperInstallConfig CRD).
+Also the default gatekeeper webhook configuration is using "failurePolicy​: ​Ignore", we assume this would be extended to the mutation feature (more on this: https://github.com/open-policy-agent/gatekeeper#admission-webhook-fail-open-status). Any resources which would slip into the system unmodified would later be corrected using the additional relabeling feature (implemented in the second phase).
+
+Currently we can only specify the namespaces to be included. The operator would have to update the webhook config every time a new namespace is added. Webhook specific exludes in kubernetes are discussed [here|https://github.com/kubernetes/kubernetes/issues/92157]
```

Comment:
this would mean restarts of gatekeeper on every removal or creation of a namespace?

Reply:
This would only update the namespaceselector of the MutatingWebhookConfiguration, the gatekeeper pods would not be touched.

Reply:
Namespace selectors cannot grow infinitely. How many namespaces do you expect? 10? 100? 1000? 10000?

Reply:
The namespace selector follows the label selector pattern, so filtering by a control-plane label would be enough. I updated the doc to be more clear on this.